common/meta/raft-store/src/state_machine/sm.rs

```
@@ -559,6 +576,265 @@ impl StateMachine {
+    ) -> bool {
+        let key = cond.key.clone();
+
+        let sub_tree = txn_tree.key_space::<GenericKV>();
+        let sv = sub_tree.get(&key);
+        if sv.is_err() {
+            return false;
+        }
+        let sv = sv.unwrap();
+        tracing::debug!("txn_execute_one_condition: {:?} {:?}", key, sv);
+
+        match sv {
+            Some(sv) => {
+                if cond.target == ConditionTarget::Value as i32 {
+                    // compare the value
+                    self.return_value_condition_result(cond.expected, cond, sv)
+                } else {
+                    // compare the seq
+                    self.return_seq_condition_result(cond.expected, cond, sv)
+                }
+            }
+            None => false,
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip(self, txn_tree))]
+    fn txn_execute_condition(
+        &self,
+        txn_tree: &TransactionSledTree,
+        condition: &Vec<TxnCondition>,
+    ) -> bool {
+        for cond in condition {
+            if !self.txn_execute_one_condition(txn_tree, cond) {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    fn txn_execute_get_operation(
+        &self,
+        txn_tree: &TransactionSledTree,
+        get: &TxnGetRequest,
+        resp: &mut TxnReply,
+    ) {
+        let key = get.key.clone();
+
+        let sub_tree = txn_tree.key_space::<GenericKV>();
+        let sv = sub_tree.get(&key);
+
+        let get = match sv {
+            Ok(v) => match v {
+                Some(v) => TxnGetResponse {
+                    key: get.key.clone(),
+                    value: convert_seqv_to_pb(Some(v)),
+                },
+                None => TxnGetResponse {
+                    key: get.key.clone(),
+                    value: None,
+                },
+            },
+            Err(_v) => TxnGetResponse {
+                key: get.key.clone(),
+                value: None,
+            },
+        };
```

Never ignore an error

---

done